csrc/ir/utils.cpp

```diff
@@ -801,6 +801,166 @@ std::vector<TensorView*> getTVsWithDynamicTransform(Fusion* fusion) {
+  std::copy_if(
+      reference.begin(),
+      reference.end(),
+      std::back_inserter(unreachable_reference_ids),
+      [&](const auto reference_id) {
+        return produced_ids.find(reference_id) == produced_ids.end();
+      });
+
+  // Check unused IDs, which should be either redundant or additional
+  // IDs.
+  std::vector<IterDomain*> unused_ids;
+  unused_ids.reserve(domain_dedup.size());
+  std::copy_if(
+      domain_dedup.begin(),
+      domain_dedup.end(),
+      std::back_inserter(unused_ids),
+      [&](const auto id_to_check) {
+        return consumed_ids.find(id_to_check) == consumed_ids.end();
+      });
+
+  std::vector<IterDomain*> additional_ids;
+
+  // If there're unused IDs, they may be redundant or additional
+  // IDs. The latter means the ID has no overlap with any other
+  // ID. This could happen, for example, when we use a concrete loop
+  // ID for a broadcast logical ID.
+  //
+  // If there's unreachable reference IDs, however, the unused IDs may have
+  // been unused because of missing dependencies, which should not be
+  // considered redundant or additional. Different analysis would be
+  // required in that case, which is not imlemented since error
+  // reporting of this function is best effort.
+  if (!unused_ids.empty() && unreachable_reference_ids.empty()) {
+    // In order to understand if an unused ID is redundant or just an
+    //  additional ID, check if the unused ID is connected to any of
+    //  the reference domain. If it's connected even with missing
+    //  dependencies, it should be considered redundant. For this
+    //  reason, a variant of IRBFS with a relaxed dependency condition
+    //  is used. IRBFSWithPermissiveDependence can traverse as long as one of
+    //  the inputs or outputs is visited.
+    const auto from_remaining_ids =
+        getExprsBetween<IRBFSWithPermissiveDependence>(
+            {unused_ids.begin(), unused_ids.end()},
+            {reference.begin(), reference.end()},
+            /*require_all_to_visited=*/false)
+            .first;
+    // Nothing is reachable, which means all of the unused IDs are not redundant
+    if (from_remaining_ids.empty()) {
+      additional_ids = unused_ids;
+    } else {
+      // Inputs of the path are redundant, and the rest are not
+      auto inputs =
+          getInputsOfExprPath(from_remaining_ids, IRInputs(), IROutputs());
+      for (const auto inp : inputs) {
+        redundant_ids.push_back(inp->as<IterDomain>());
+      }
+      for (const auto unused_id : unused_ids) {
+        if (std::find(inputs.begin(), inputs.end(), unused_id) ==
+            inputs.end()) {
+          additional_ids.push_back(unused_id);
+        }
+      }
+    }
+  }
+
+  // If an output of the path is not used, i.e., not part of the
+  // reference domain, it's considered an additional ID. For example,
+  // if we have `split i0 -> i1, i2`, and the reference only includes
+  // i1, i2 is considered an additional ID
```

Comment:
For clarity:
```suggestion
  // i1, then i2 is considered an additional ID
```
So that we now it is not "includes both i1 and i2"